### PR TITLE
chore: adding the docker vulnerability monitor

### PR DESCRIPTION
**Pull request by chenbishop**
## Related Tickets & Documents

- Fixes https://github.com/Twingate/kubernetes-operator/issues/26

## Changes

- adding the docker image vulnerability monitor and send Slack message to defined Slack channel if fixable vulnerability is found
-  the workflow is ran daily

### CODE REVIEW

File: .github/workflows/vulnerability-monitor.yaml [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+  docker-image-monitor:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Login to Docker Hub
+        uses: docker/login-action@v3
+        with:
+          username: ${{ secrets.DOCKERHUB_USERNAME }}
+          password: ${{ secrets.DOCKERHUB_TOKEN }}
+      - run: |
+          sudo apt-get update
```
Comment (ekampf) [inline .github/workflows/vulnerability-monitor.yaml:17]: Why not https://github.com/docker/scout-action ?

**Review by ekampf — CHANGES_REQUESTED** [review]
see comment

File: .github/workflows/vulnerability-monitor.yaml (hunk 1 above)
Comment (chenbishop) [reply .github/workflows/vulnerability-monitor.yaml:17]: I tried it on my local, and it wasnt working, stuck on setting up the scout-action while trying to clone the repo due to binaries stored in repo https://github.com/docker/scout-action/tree/main/dist.

Comment (chenbishop) [reply .github/workflows/vulnerability-monitor.yaml:17]: @ekampf, @emrul  and I looked at this in more details and we think there is also a security consideration. The scout action includes a binary and we cannot see the source code for this binary, see this commit as an example https://github.com/docker/scout-action/commit/80fc6e99dd73c7d09db18aa7d4006bebb3f5c11a#diff-a74d462be054c5a1a941b83a506392c5c7caf269aeea819f1d7d13c483a41f93

Comment (ekampf) [reply .github/workflows/vulnerability-monitor.yaml:17]: Its the **official docker action** not some binary from a random person on the internet

File: .github/workflows/vulnerability-monitor.yaml [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+          sudo apt-get install jq
+          curl -sSfL https://raw.githubusercontent.com/docker/scout-cli/main/install.sh | sh -s --
+          docker scout cves twingate/kubernetes-operator:dev --only-fixed --format sarif --output kubernetes-operator.sarif.json
+          if [ `cat kubernetes-operator.sarif.json | jq -r '.runs[0].results | length'` -gt 0 ];
+          then docker scout cves twingate/kubernetes-operator:dev --only-fixed --output kubernetes-operator.txt;
```
Comment (ekampf) [inline .github/workflows/vulnerability-monitor.yaml:22]: 1. Code isn't properly indented
2. You dont need to run docker scout twice and use `jq` to check if there are only error you can check exit code:
```
  -e, --exit-code                   Return exit code '2' if vulnerabilities are detected
```
3. lets add `--only-severity` for medium\high\critical

**Review by ekampf — APPROVED** [review]

**Pull request merged**